docs/intro/overview.md

```diff
@@ -1,61 +1,70 @@
+The VPP agent configuration behavior is as follows:
 
-Another significant feature is ability to retrieve existing VPP configuration. The retrieved data is used not only for status reporting, but also for resynchronization during which it is compared to desired configuration to resolve required changes and minimize the impact during restarts.
+* Configuration items with all dependencies satisfied or no dependencies are programmed into VPP
 
-We also often stumbled upon the case where configured VPP worked as expected, and after restart/reconfiguration the VPP state looked exactly the same - but did not work as before. The reason for this behavior is that the binary API calls followed incorrect order during restart, configuring the VPP only ostensibly correct. This can be also solved by the Agent - plugins ensure all API calls are called in the correct sequence in order to make it work right.
+* Incomplete configuration items (those with unresolved dependencies) are cached or if possible partially completed
 
-#### Resync mechanism
+*  At some later point in time, when the missing dependency items appear, all pending configuration items are re-validated and executed. All configuration items including those with dependencies between VPP and Linux is managed the same way.
 
-The resync is one of the major Agent features - it ensures consistency between configuration provided from an external source, internal Agent state and the actual VPP state. The automatic resync fetches all the data from any connected persistent data store and reflects changes to the VPP. The synchronization is also performed against the Linux host. 
-The resync is by default started on the Agent startup, but can also automatically be launched on certain events (VPP restart, reconnection to the data base). 
+Another significant feature is the ability to retrieve existing VPP configuration. In addition to status reporting, this data is used to perform resynchronization. This is the process by which the active VPP configuration is compared to the desired VPP configuration to resolve any required changes and to minimize the impact during restarts.
 
-#### Plugin concept
+```
+The need to address the configuration dependency problem arose when a configured VPP worked as expected. Following a restart/reconfiguration, the VPP state looked exactly the same - but did not work as before. The reason for this behavior is that the binary API calls followed an incorrect order during restart, resulting in a partially complete configuration and VPP dataplane errors.
```

Comment:
Why is this highlighted by triple backtick? It's not a code block - perhaps use [admonition note/important/whatever notation](https://squidfunk.github.io/mkdocs-material/extensions/admonition/) instead.